test/common/buffer/owned_impl_test.cc

```diff
@@ -1166,7 +1218,7 @@ TEST_F(OwnedImplTest, ReserveZeroCommit) {
   ASSERT_EQ(os_sys_calls.close(pipe_fds[1]).rc_, 0);
   ASSERT_EQ(previous_length, buf.search(data.data(), rc, previous_length, 0));
   EXPECT_EQ("bbbbb", buf.toString().substr(0, 5));
-  expectSlices({{5, 0, 4096}, {1953, 2143, 4096}}, buf);
+  expectSlices({{5, 0, 4096}, {1953, 14431, 16384}}, buf);
```

Comment:
There seems to be some subtlety in this test case which is not fully captured by test expectations.

A potential improvement would be to add an expectSlices before the reserveSingleSlice call to capture the state of the buffer.  I think we will see something like:
expectSlices({{5, 0, 4096}, {0, 0, 0}}, buf);
or
expectSlices({{5, 0, 4096}, {0, 0, 0}, {0, 4096, 4096}}, buf);

That said, I'm not sure how relevant this test is given that it refers to libevent buffer implementation which was removed long ago.

Reply:
Adding some extra expectations to make it clear what happens in each step of this test. It shows that two operations that should really be no-ops both changed the buffer, although in unimportant ways. This is a candidate for some cleanup in a later PR (`add("")` probably shouldn't create a zero-length slice; it should be a no-op).